src/middy.js

```diff
@@ -72,7 +72,7 @@ const runErrorMiddlewares = (middlewares, instance, done) => {
   const runNext = (err) => {
     try {
       if (!err) {
-        return done()
+        instance.__handledError = true
```

Comment:
Good point. It's not even easy to explain.

What happens with this new logic is that, in case of error, the execution is moved to the error middlewares stack. In the previous version, as soon as one error middleware handled the error (by not propagating the error in the `next()` function), the execution stopped and the response was returned.

Now we thought it makes more sense to keep executing all the other error middlewares so that they have a chance to keep decorating the response (e.g. add headers, serialize the body, etc.)

With this new approach we need a way to figure out if the error was somehow handled so that we can decide wether the execution global execution of the lambda needs to be considered failed (unhandled error) or successful (handled error). We use that flag to do that